cardano-launcher/app/Main.hs

```diff
@@ -39,22 +40,30 @@ import           Data.Text.Lazy.Builder (fromString, fromText)
 -- | Main function.
 main :: IO ()
 main = do
+    logConfig       <- defaultConfigStdout
+    (baseTrace, sb) <- setupTrace_ logConfig "launcher"
+
+    Trace.logNotice baseTrace "Starting cardano-launcher"
 
     setEnv "LC_ALL" "en_GB.UTF-8"
     setEnv "LANG"   "en_GB.UTF-8"
 
     launcherOptions <- do
         eLauncherOptions <- getLauncherOptions
         case eLauncherOptions of
-            Left err -> throwM $ LauncherOptionsError (show err)
+            Left err -> do
+                Trace.logError baseTrace $
```

Comment:
It's good like this, but you could have a special function called `logAndThrowM` which does both things and return both error messages equally.